controller/linodemachine_controller_helpers.go

```diff
@@ -109,20 +109,30 @@ func (r *LinodeMachineReconciler) newCreateConfig(ctx context.Context, machineSc
 func (r *LinodeMachineReconciler) buildInstanceAddrs(ctx context.Context, machineScope *scope.MachineScope, instanceID int) ([]clusterv1.MachineAddress, error) {
 	addresses, err := machineScope.LinodeClient.GetInstanceIPAddresses(ctx, instanceID)
 	if err != nil {
-		return nil, fmt.Errorf("get instance ips: %w", err)
+		return nil, err
 	}
 
 	// get the default instance config
 	configs, err := machineScope.LinodeClient.ListInstanceConfigs(ctx, instanceID, &linodego.ListOptions{})
 	if err != nil || len(configs) == 0 {
-		return nil, fmt.Errorf("list instance configs: %w", err)
+		return nil, err
 	}
 
 	ips := []clusterv1.MachineAddress{}
-	// check if a node has public ip and store it
-	if len(addresses.IPv4.Public) != 0 {
-		ips = append(ips, clusterv1.MachineAddress{Address: addresses.IPv4.Public[0].Address, Type: clusterv1.MachineExternalIP})
+	// check if a node has public ipv4 ip and store it
+	if len(addresses.IPv4.Public) == 0 {
+		return nil, fmt.Errorf("No Public IPv4 address set on machine")
```

Comment:
If you do not need a formatted error message, please use `errors.New` instead:

```suggestion
		return nil, errors.New("no public ipv4 addresses set on machine")
```

Reply:
This could probably also be hoisted up into being a sentinel error:

```go

var errNoPublicIPv4Addrs = errors.New("no public ipv4 addresses set")